shim/stub.go

```diff
@@ -583,12 +623,16 @@ func (s *ChaincodeStub) GetQueryResultWithPagination(query string, pageSize int3
 
 // StartWriteBatch documentation can be found in interfaces.go
 func (s *ChaincodeStub) StartWriteBatch() {
-	s.handler.handleStartWriteBatch(s.ChannelID, s.TxID)
+	if s.writeBatch == nil {
+		s.writeBatch = newWriteBatch()
+	}
```

Comment:
I don't see a check for the usePeerWriteBatch flag.

Reply:
```
if s.handler.usePeerWriteBatch && s.writeBatch == nil {
	s.writeBatch = newWriteBatch()
}
```

Reply:
Good catch. It's a shame the unit tests don't check this.